### PR TITLE
tests: tests for NewMultipart Upload HTTP handler.

### PR DESCRIPTION
**Pull request by hackintoshrao**

### CODE REVIEW

File: cmd/object-handlers_test.go [hunk 1]
```diff
@@ -438,3 +440,171 @@ func testAPICopyObjectHandler(obj ObjectLayer, instanceType, bucketName string,
...
+		}
+		// compare the uploadID obtained from response with uploadID obtained from parsing upload.json .
+		// Assert the number of uploadID entries for the object in upload.json .
+
+		if len(testUploads.uploads) != len(uploadsJSON.Uploads) {
+			t.Fatalf("Minio %s: Expected record of %d uploadID's in upload.json for given object, but got %d.", instanceType, len(testUploads.uploads), len(uploadsJSON.Uploads))
+		}
+		// Entry for the upload ID's generated from the response should be found in upload.json .
+
+	CheckXlUploadID:
```
Comment (harshavardhana) [inline cmd/object-handlers_test.go:573]: ??? - we should avoid these..  is this really needed?

File: cmd/object-handlers_test.go [hunk 2]
```diff
@@ -438,3 +440,171 @@ func testAPICopyObjectHandler(obj ObjectLayer, instanceType, bucketName string,
...
+		if err != nil {
+			t.Fatalf("Reading uploads json from FS backend failed: <ERROR> %s.", err)
+		} // compare the uploadID obtained from response with uploadID obtained from parsing upload.json .
+		// Assert the number of uploadID entries for the object in upload.json .
+
+		if len(testUploads.uploads) != len(uploadsJSON.Uploads) {
+			t.Fatalf("Minio %s:Expected record of %d uploadID's in upload.json for given object, but got %d.", instanceType, len(testUploads.uploads), len(uploadsJSON.Uploads))
+		}
+		// Entry for the upload ID's generated from the response should be found in upload.json .
+	CheckFsUploadID:
```
Comment (harshavardhana) [inline cmd/object-handlers_test.go:596]: ??? - we should avoid these..  is this really needed?

File: cmd/object-handlers_test.go [hunk 3]
```diff
@@ -438,3 +440,171 @@ func testAPICopyObjectHandler(obj ObjectLayer, instanceType, bucketName string,
...
+	// Wait till all go routines finishes execution.
+	wg.Wait()
+
+	// verify the existance and correctness of upload.json.
+	var uploadsJSON uploadsV1
+	var err error
+	switch v := obj.(type) {
+	case xlObjects:
+		// read the upload.json for xl backend.
+		uploadsJSON, err = v.readUploadsJSON(bucketName, objectName)
```
Comment (harshavardhana) [inline cmd/object-handlers_test.go:561]: Calling this handler test we shouldn't use internal functions.. this is a bad idea and hard to extend these tests.  What is the purpose here?

File: cmd/object-handlers_test.go [hunk 4]
```diff
@@ -438,3 +440,171 @@ func testAPICopyObjectHandler(obj ObjectLayer, instanceType, bucketName string,
...
+	err = decoder.Decode(multipartResponse)
+	if err != nil {
+		t.Fatalf("Error decoding the recorded response Body")
+	}
+	// verify the existance and correctness of upload.json.
+	var uploadsJSON uploadsV1
+	switch v := obj.(type) {
+	case xlObjects:
+		// read the upload.json for xl backend.
+		uploadsJSON, err = v.readUploadsJSON(bucketName, objectName)
```
Comment (harshavardhana) [inline cmd/object-handlers_test.go:482]: We should only validate indeed if the response was correct we don't need to see if the backend was read properly.  In this situation we are skipping locking and all other forms of sanity checks leading to a bad test.

File: cmd/object-handlers_test.go (hunk 1 above)
Comment (hackintoshrao) [reply cmd/object-handlers_test.go:573]: Need to break away from outer loop too.

File: cmd/object-handlers_test.go (hunk 4 above)
Comment (hackintoshrao) [reply cmd/object-handlers_test.go:482]: Will remove validation of upload.json .

**Comment by hackintoshrao** [comment]
Closing the PR, will be making it a short tests without the upload.json assertions, Will send it along with other Multipart handler tests.

**Pull request closed without merging**